src/js/program/media-element-pool.ts

```diff
@@ -20,30 +36,31 @@ export default function MediaElementPool() {
     let primed = false;
 
     return {
-        primed() {
+        primed(): boolean {
             return primed;
         },
-        prime() {
+        prime(): void {
             elements.forEach(primeMediaElementForPlayback);
             primed = true;
         },
-        played() {
+        played(): void {
             primed = true;
         },
-        getPrimedElement() {
-            if (pool.length) {
+        getPrimedElement(): HTMLVideoElement | null {
+            const el = pool.shift();
+            if (el) {
                 // Shift over pop so that we cycle through elements instead of reusing the same one
-                return pool.shift();
+                return el;
             }
             return null;
         },
-        getAdElement() {
+        getAdElement(): OptionalUndefVideo {
             return adElement;
         },
-        getTestElement() {
+        getTestElement(): OptionalUndefVideo {
```

Comment:
testElement must always be defined.

Reply:
same comment re: `pool.shift`

Reply:
Use `as` to assign or if we don't need the element to come from the pool, then break out the element factory.

Reply:
Just make sure we don't do anything with the pool before pulling out test and ad elements (like "priming" the tags).